Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +161,34 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+            // When array == null return number
+            if (array == null) return new int[1] { number };
+            if (array.Length == 0)
+            {
+                if (index < 0 || index > array.Length) return array;
```

Comment:
Is there a simpler way to express the meaning behind `index < 0 || index > array.Length`? Can you combine that with the `if` on line 166 to reduce the branch nesting?